src/cryptography/hazmat/backends/openssl/aead.py

```diff
@@ -13,11 +13,18 @@
 
 def _aead_cipher_name(cls, key_length):
```

Comment:
Ok, rather than pass the type here and doing all this weird `if key_lenght is None` stuff, let's pass an instance of the aead instead of the `cls`, then we can use the `key` attr.